pkg/tfbridge/provider_test.go

```diff
@@ -756,3 +706,288 @@ func TestProviderReadNestedSecretV2(t *testing.T) {
 
 	testProviderReadNestedSecret(t, provider, "NestedSecretResource")
 }
+
+func TestCheckConfig(t *testing.T) {
+	t.Run("minimal", func(t *testing.T) {
+		// Ensure the method is minimally implemented. Pulumi will be passing a provider version. Make sure it
+		// is mirrored back.
+		provider := &Provider{
+			tf:     shimv2.NewProvider(testTFProviderV2),
+			config: shimv2.NewSchemaMap(testTFProviderV2.Schema),
+		}
+		testutils.Replay(t, provider, `
+		{
+		  "method": "/pulumirpc.ResourceProvider/CheckConfig",
+		  "request": {
+		    "urn": "urn:pulumi:dev::teststack::pulumi:providers:testprovider::test",
+		    "olds": {},
+		    "news": {
+		      "version": "6.54.0"
+		    }
+		  },
+		  "response": {
+		    "inputs": {
+		      "version": "6.54.0"
+		    }
+		  }
+		}`)
+	})
+
+	t.Run("config_value", func(t *testing.T) {
+		provider := &Provider{
+			tf:     shimv2.NewProvider(testTFProviderV2),
+			config: shimv2.NewSchemaMap(testTFProviderV2.Schema),
+		}
+		// Ensure Pulumi can configure config_value in the testprovider.
+		testutils.Replay(t, provider, `
+		{
+		  "method": "/pulumirpc.ResourceProvider/CheckConfig",
+		  "request": {
+		    "urn": "urn:pulumi:dev::teststack::pulumi:providers:testprovider::test",
+		    "olds": {},
+		    "news": {
+                      "config_value": "foo",
+		      "version": "6.54.0"
+		    }
+		  },
+		  "response": {
+		    "inputs": {
+                      "config_value": "foo",
+		      "version": "6.54.0"
+		    }
+		  }
+		}`)
+	})
+
+	t.Run("config_changed", func(t *testing.T) {
+		provider := &Provider{
+			tf:     shimv2.NewProvider(testTFProviderV2),
+			config: shimv2.NewSchemaMap(testTFProviderV2.Schema),
+		}
+		// In this scenario Pulumi plans an update plan when a config has changed on an existing stack.
+		testutils.Replay(t, provider, `
+		{
+		  "method": "/pulumirpc.ResourceProvider/CheckConfig",
+		  "request": {
+		    "urn": "urn:pulumi:dev::teststack::pulumi:providers:testprovider::test",
+		    "olds": {
+                      "config_value": "foo",
+		      "version": "6.54.0"
+                    },
+		    "news": {
+                      "config_value": "bar",
+		      "version": "6.54.0"
+		    }
+		  },
+		  "response": {
+		    "inputs": {
+                      "config_value": "bar",
+		      "version": "6.54.0"
+		    }
+		  }
+		}`)
+	})
+
+	// TODO[pulumi/pulumi-terraform-bridge#244] check validation scenarios once CheckConfig re-enables validation.
```

Comment:
Is this still TODO? It looks like `CheckConfig` is performing validation in this change.

Reply:
Yeah let me go and fill this test out. To check the invalid setting paths.